esb/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/BAMMediatorTransformer.java

```diff
@@ -58,7 +58,7 @@ public void transformWithinSequence(TransformationInfo information, EsbNode subj
 		/*
 		 * Check subject.
 		 */
-		Assert.isTrue(subject instanceof BAMMediator, "Invalid subject.");
+		Assert.isTrue(subject instanceof BAMMediator, "Invalid subject."); //$NON-NLS-1$
```

Comment:
Why do we need this Assert operation? Using Assert to decide programming flow is not good practice. Instead of this, can we use simple if ladder to check the conditions?